src/Serialization/main.jl

```diff
@@ -1,9 +1,135 @@
+################################################################################
+# Description of the saving and loading mechanisms
+#
+# We require that any types serialized through OSCAR are registered using the
+# @register_serialization_type macro, like so
+
+# @register_serialization_type NewType "String Representation of type" uses_id uses_params
+
+# The last three arguments are optional and can arise in any order. Passing a String
+# arguement will override how the type is store as a string. The last two are boolean
```

Comment:
```suggestion
# argument will override how the type is stored as a string. The last two are boolean
```
I think the explanation for the optional string needs to be extended. How about this:

> Each serializable type is associated to a unique id that is written and read during initialization. By default this identifier is `string(NewType)` but in some cases it can be desirable to customize it. This is what the optional string argument does, it overrides the id.